tests/test_report.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+
+from . import v
+assert v  # Silence pyflakes
+
+mock_code = """\
+import foo
+
+class Foo:
+    '''
+    This is a long class
+    '''
+    def __init__(self):
+        print("Initialized foo")
+
+    def bar():
+        foobar = "unused variable"
+        return
+        print("A small unused function")
+"""
+
+
+@pytest.fixture
+def check_report(v, capsys, tmpdir):
+    def test_report(code, expected, min_confidence=0, sort_by_size=False,
+                    make_whitelist=False):
+        filename = str(tmpdir.join('dead_code.py'))
+        with open(filename, 'w') as f:
```

Comment:
Do we really need temporary files to test the new feature?

Reply:
I guess, it makes tests pretty robust since tests will break if any part of the pipeline is tampered with. But, still I'd say it's your call.

Reply:
I forgot that `Vulture.make_whitelist()` directly writes to stdout.